db/attachment_compaction.go

```diff
@@ -551,3 +559,12 @@ func getCompactionDCPClientOptions(collection *base.Collection, groupID string)
 	return clientOptions, nil
 
 }
+
+func generateCompactionDCPStreamName(compactionID, compactionAction string) string {
```

Comment:
Have you seen / considered using `base.GenerateDcpStreamName`? Even if it needs changing to work with attachment compaction, it might still be worthwhile instead of a new function that does something very similar.

Reply:
This would require big changes to that function, and would mean making two code paths as well as arguments needing to be passed in that may not be used depending on the code path and would get quite messy. I think it would be best to have this change separate to keep the code clean, and the attachment compaction functions all together.

Reply:
Agreed, trying to reuse that function wouldn't be clean. Even stuff like the UUID generation can't be used as we need to pass in the compaction ID. You'd basically be passing everything down at that point.